src/App.tsx

```diff
@@ -1,39 +1,76 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
+
+type State = {
+  hasClock: boolean,
+  clockName: string,
+  timerId: number,
+};
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends React.Component<{}, State> {
+  state = {
+    hasClock: true,
+    clockName: 'Clock-0',
+    timerId: 0,
+  };
+
+  today = new Date();
+
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.hideClock, true);
+    document.addEventListener('click', this.showClock, true);
+    this.state.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidUpdate(_prevProps: {}, prevState: Readonly<State>) {
+    if (prevState.clockName !== this.state.clockName && this.state.hasClock) {
+      console.debug(`Renamed from ${prevState.clockName} to ${this.state.clockName}`);
```

Comment:
you can add this code before console statement to prevent linter error
`// eslint-disable-next-line no-console`